File: tests/golang/testC.go
package main

import "fmt"

func printHelloTestC1() {
    fmt.Println("Hello, World!")
    for i := 0; i < 5; i++ {
        fmt.Printf("This is line %d\n", i)
        if i%2 == 0 {
            fmt.Println("Even number")
        } else {
            fmt.Println("Odd number")
        }
    }
    fmt.Println("End of printHello")
}

func printHelloTestC2() {
    fmt.Println("Hello, World!")
    for i := 0; i < 5; i++ {
        fmt.Printf("This is line %d\n", i)
        if i%2 == 0 {
            fmt.Println("Even number")
        } else {
            fmt.Println("Odd number")
        }
        fmt.Printf("Current iteration: %d\n", i)
    }
    fmt.Println("End of printHelloAgain")
}

func main() {
    printHelloTestC1()
    printHelloTestC2()
}

File: tests/rust/testC.rs
fn print_hello_test_c1() {
    println!("Hello, World!");
    for i in 0..5 {
        println!("This is line {}", i);
        if i % 2 == 0 {
            println!("Even number");
        } else {
            println!("Odd number");
        }
    }
    println!("End of printHello");
}

fn print_hello_test_c2() {
    println!("Hello, World!");
    for i in 0..5 {
        println!("This is line {}", i);
        if i % 2 == 0 {
            println!("Even number");
        } else {
            println!("Odd number");
        }
        println!("Current iteration: {}", i);
    }
    println!("End of printHelloAgain");
}

fn main() {
    print_hello_test_c1();
    print_hello_test_c2();
}